chpl/chpl-service/src/main/java/gov/healthit/chpl/app/chartdata/IncumbentDevelopersStatisticsCalculator.java

```diff
@@ -0,0 +1,166 @@
+        from2014To2015.setOldCertificationEditionId(2L);
+        from2014To2015.setNewCertificationEditionId(3L);
+        LOGGER.info("Total 2011 Developers: " + developers2011.size());
+        LOGGER.info("Total 2014 Developers: " + developers2014.size());
+        LOGGER.info("Total 2015 Developers: " + developers2015.size());
+        for (String name : developers2014) {
+            if (developers2011.contains(name)) {
+                from2011To2014.setIncumbentCount(from2011To2014.getIncumbentCount() + 1);
+            } else {
+                from2011To2014.setNewCount(from2011To2014.getNewCount() + 1);
+            }
+        }
+        for (String name : developers2015) {
+            if (developers2011.contains(name)) {
+                from2011To2015.setIncumbentCount(from2011To2015.getIncumbentCount() + 1);
+            } else {
+                from2011To2015.setNewCount(from2011To2015.getNewCount() + 1);
+            }
+            if (developers2014.contains(name)) {
+                from2014To2015.setIncumbentCount(from2014To2015.getIncumbentCount() + 1);
+            } else {
+                from2014To2015.setNewCount(from2014To2015.getNewCount() + 1);
+            }
+        }
+        ArrayList<IncumbentDevelopersStatisticsEntity> result = new ArrayList<IncumbentDevelopersStatisticsEntity>();
+        result.add(from2011To2015);
+        result.add(from2014To2015);
+        result.add(from2011To2014);
+        return result;
+    }
+
+    private void initialize() {
+        incumbentDevelopersStatisticsDAO = (IncumbentDevelopersStatisticsDAO)
+                appEnvironment.getSpringManagedObject("incumbentDevelopersStatisticsDAO");
+        txnManager = (JpaTransactionManager) appEnvironment.getSpringManagedObject("transactionManager");
+        txnTemplate = new TransactionTemplate(txnManager);
+    }
+
+    private void logCounts(final List<IncumbentDevelopersStatisticsEntity> entities) {
+        for (IncumbentDevelopersStatisticsEntity entity : entities) {
+            LOGGER.info("Incumbent Developer statistics: [" + entity.toString() + "]");
+        }
+    }
+
+    private void save(final List<IncumbentDevelopersStatisticsEntity> entities) {
+        txnTemplate.execute(new TransactionCallbackWithoutResult() {
+
+            @Override
+            protected void doInTransactionWithoutResult(final TransactionStatus arg0) {
+                try {
+                    deleteExistingIncumbentDevelopersStatistics();
+                } catch (EntityRetrievalException e) {
+                    LOGGER.error("Error occured while deleting existing CriterionProductStatistics.", e);
+                    return;
+                }
+                try {
+                    for (IncumbentDevelopersStatisticsEntity entity : entities) {
+                        IncumbentDevelopersStatisticsDTO dto = new IncumbentDevelopersStatisticsDTO(entity);
```

Comment:
I think it's a bit repetitive here to have a list of entities, convert each one to a DTO, then in the dao create method they get converted back to an entity for saving to the db. It was always my understanding (from whenever Watson introduced me to the code, I think) that we were supposed to keep the entities used primarily only in the DAOs and everywhere else we should pass around DTO objects... maybe some of the other code in this class could be changed to use DTOs and then it would just get converted to an entity once inside the DAO?